Automattic/jetpack
# Instant Search: Improve history navigation handling, open results in current window

Fixes https://github.com/Automattic/jetpack/issues/18455.
Fixes #18555.
Fixes #18701.
Fixes #18667.
Fixes 1250-gh-Automattic/p2.

#### Changes proposed in this Pull Request:

* Open search results in the current tab rather than opening a new tab with `target="_blank"`.
* Fixes integration with browser history navigation (back/forward). For example, if you search without a query, click a result link, then hit back, the search overlay will now reappear.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
On a site with Instant Search enabled, ensure that search result links all open in the same window (including image links). When you press the back button, the search modal should be redisplayed, even if you did not originally specify a search query.

On a site with a search modal trigger (like Twenty Twenty), make sure that the search modal opens with the URL `/?s=`. Make sure this is removed when the modal is closed.

<img width="1413" alt="Screen Shot 2021-01-27 at 14 07 09" src="https://user-images.githubusercontent.com/17325/105927519-08e69200-60a9-11eb-9bfd-8ba9a0fdb1a0.png">

#### Proposed changelog entry for your changes:

* Instant Search: open search result links in the current window rather than opening a new one.
* Instant Search: improves compatibility with browser forward/back navigation.
